src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -415,7 +415,17 @@ protected Task TestSmartTagTextAsync(string initialMarkup, string displayText, i
             var fixedRoot = await document.GetSyntaxRootAsync();
             var actualText = fixedRoot.ToFullString();
 
-            Assert.Equal(expectedText, actualText);
+            // To help when a user just writes a test (and supplied no 'expectedText') just print
+            // out the entire 'actualText' (without any trimming).  in the case that we have both,
+            // call the normal Assert helper which will print out a good trimmed diff. 
+            if (expectedText == "")
```

Comment:
The comment sounds great, but I didn't understand the implementation (both branches seem identical).

Reply:
One calls the (object, object) overload, one calls the (string, string) overload.  On the latter has hte special string-diffing logic.  The former does a much more basic compare-and-print-if-different.